dev/MenuBar/MenuBarItem.cpp

```diff
@@ -260,13 +260,41 @@ void MenuBarItem::OpenFlyoutFrom(FlyoutLocation location)
         CloseMenuFlyout();
         if (location == FlyoutLocation::Left)
         {
-            winrt::get_self<MenuBarItem>(menuBar.Items().GetAt(((index - 1) + menuBar.Items().Size()) % menuBar.Items().Size()))->ShowMenuFlyout();
+            if (const auto item = GetNextItem(index, -1).try_as<MenuBarItem>())
+            {
+                item->ShowMenuFlyout();
+            }
         }
         else
         {
-            winrt::get_self<MenuBarItem>(menuBar.Items().GetAt((index + 1) % menuBar.Items().Size()))->ShowMenuFlyout();
+            if (const auto item = GetNextItem(index, +1).try_as<MenuBarItem>())
+            {
+                item->ShowMenuFlyout();
+            }
+        }
+    }
+}
+
+winrt::MenuBarItem MenuBarItem::GetNextItem(int index, int direction)
+{
+    if (const auto menuBar = m_menuBar.get())
+    {
+        const int itemsCount = menuBar.Items().Size();
+        // % is remainder, not modulo.
```

Comment:
>modulo [](start = 31, length = 6)

What does this comment mean?

Reply:
Index + direction could be negative, using `% itemscount` would give us a negative index which is invalid. A lot of programming languages treat % as modulo (mapping any number from 0 to n-1), C++ does not though so we need to add itemscount to ensure we are ending up with a positive index.

Reply:
Sure, I'm not sure we need a comment explaining how c++ works though. Maybe the comment could be "itemIndex+direction could be negative"

---
In reply to: [522503808](https://github.com/microsoft/microsoft-ui-xaml/pull/3555#discussion_r522503808) [](ancestors = 522503808)

Reply:
Good point, updated the comments to point that out instead of the remainder/modulo thing.